2009-11-22  Aurelien Jarno  <aurelien@aurel32.net>

	* nis/nss_nis/nis-hosts.c (_nss_nis_gethostbyname4_r): If
	NIS is not configured, set errnop and herrnop appropriately.

--- a/nis/nss_nis/nis-hosts.c
+++ b/nis/nss_nis/nis-hosts.c
@@ -260,7 +260,11 @@
 
   char *domain;
   if (yp_get_default_domain (&domain))
-    return NSS_STATUS_UNAVAIL;
+    {
+      *h_errnop = HOST_NOT_FOUND;
+      *errnop = errno;
+      return NSS_STATUS_UNAVAIL;
+    }
 
   if (buflen < sizeof *data + 1 + pad)
     {
